src/AggregateFunctions/Combinators/AggregateFunctionDistinct.h

```diff
@@ -176,12 +212,16 @@ class AggregateFunctionDistinct final : public IAggregateFunctionDataHelper<Data
 
     void add(AggregateDataPtr __restrict place, const IColumn ** columns, size_t row_num, Arena * arena) const override
     {
-        this->data(place).add(columns, arguments_num, row_num, arena);
+        auto argument_columns = prepareArgumentColumns();
```

Comment:
I'm afraid this is slow.

Why don't we check the existence in the set, and if the element doesn't exist, reuse the `columns` to add it to the nested data?